src/org/opendatakit/briefcase/ui/pull/PullPanel.java

```diff
@@ -63,11 +63,15 @@ public PullPanel(TransferPanelForm view, TransferForms forms, BriefcasePreferenc
     getContainer().addComponentListener(analytics.buildComponentListener("Pull"));
 
     // Read prefs and load saved remote server if available
-    this.source = RemoteServer.readPreferences(tabPreferences).flatMap(view::preloadSource);
-    this.source.ifPresent(source -> {
-      forms.load(source.getFormList());
-      view.refresh();
-      updateActionButtons();
+    source = RemoteServer.readPreferences(tabPreferences).flatMap(view::preloadSource);
```

Comment:
Is this still happenning? I think `SourcePanel.preload(RemoteServer server)` deals with this:
```java
public Optional<Source<?>> preload(RemoteServer server) {
  try {
    Source source = getSource(server);
    source.set(server);
    triggerOnSource(source);
    return Optional.of(source);
  } catch (HttpException e) {
    log.warn("Can't preload source. Resetting view", e);
    reset();
    return Optional.empty();
  }
}
```

Reply:
I think we need to review this whole workflow, though. It could be easier to follow or, at least have some comments that explain the sequence of events/invocations.

Reply:
After looking a little more into this, it turns out that the behavior is not consistent depending on what happens with the HTTP request:
- If it timeouts, it will throw an `HttpException` and the change on this PR will prevent a program failure
- If the server responds with any non-success (2xx) status code, it won't fail but it will show an empty form list, which is really confusing.

Let's leave your `try/catch` but work a little bit more the `catch` part.
I'll open a new issue to remember we need to normalize the behavior when we get a `4xx` or a `5xx`.

Reply:
After looking a little more into this, it turns out that the behavior is not consistent depending on what happens with the HTTP request:
- If it timeouts, it will throw an `HttpException` and the change on this PR will prevent a program failure
- If the server responds with any non-success (2xx) status code, it won't fail but it will show an empty form list, which is really confusing.

Let's leave your `try/catch` but work a little bit more the `catch` part.
I'll open a new issue to remember we need to normalize the behavior when we get a `4xx` or a `5xx`.